exp/addons/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -0,0 +1,129 @@
+type ClusterResourceSetBindingReconciler struct {
+	Client client.Client
+	Log    logr.Logger
+
+	scheme *runtime.Scheme
+}
+
+func (r *ClusterResourceSetBindingReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&addonsv1.ClusterResourceSetBinding{}).
+		Watches(
+			&source.Kind{Type: &clusterv1.Cluster{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.clusterToClusterResourceSetBinding)},
+		).
+		WithOptions(options).
+		WithEventFilter(predicates.ResourceNotPaused(r.Log)).
+		Build(r)
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	r.scheme = mgr.GetScheme()
+	return nil
+}
+
+func (r *ClusterResourceSetBindingReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
+	ctx := context.Background()
+
+	// Fetch the ClusterResourceSetBinding instance.
+	crsBinding := &addonsv1.ClusterResourceSetBinding{}
+	if err := r.Client.Get(ctx, req.NamespacedName, crsBinding); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	clusterOwner := getOwnerRef(crsBinding.OwnerReferences)
+
+	cluster := &clusterv1.Cluster{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: clusterOwner.Name, Namespace: crsBinding.Namespace}, cluster); err != nil {
+		if apierrors.IsNotFound(err) || !cluster.DeletionTimestamp.IsZero() {
+			return r.reconcileDelete(ctx, crsBinding)
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// reconcileDelete handles ClusterResourceSetBinding deletion.
+func (r *ClusterResourceSetBindingReconciler) reconcileDelete(ctx context.Context, crsBinding *addonsv1.ClusterResourceSetBinding) (reconcile.Result, error) {
+	err := r.Client.Delete(ctx, crsBinding)
+	return ctrl.Result{}, err
+}
+
+// getOwnerRef returns the owner reference of type Cluster
+func getOwnerRef(ownerReferences []metav1.OwnerReference) *metav1.OwnerReference {
+	for _, r := range ownerReferences {
+		if r.Kind == "Cluster" {
+			return &r
+		}
+	}
+	return nil
+}
+
+// clusterToClusterResourceSetBinding is mapper function that maps clusters to ClusterResourceSetBinding
+func (r *ClusterResourceSetBindingReconciler) clusterToClusterResourceSetBinding(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	cluster, ok := o.Object.(*clusterv1.Cluster)
+	if !ok {
+		r.Log.Error(nil, fmt.Sprintf("Expected a Cluster but got a %T", o.Object))
+		return nil
+	}
+
+	crsBinding := &addonsv1.ClusterResourceSetBinding{}
+	if err := r.Client.Get(context.Background(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, crsBinding); err != nil {
+		r.Log.Error(err, "failed to get ClusterResourceSetBinding")
```

Comment:
```suggestion
		if !apierrors.IsNotFound(err) {
			r.Log.Error(err, "failed to get ClusterResourceSetBinding")
		}
```

Because this is triggered for every cluster update, and not all clusters have bindings, we should skip logging on 404s.